datacube_stats/tasks.py

```diff
@@ -23,23 +23,25 @@
 
 
 def select_task_generator(input_region, storage, filter_product):
+    _LOG.debug("input region %s", input_region)
     if input_region is None or input_region == {}:
         _LOG.info('No input_region specified. Generating full available spatial region, gridded files.')
         return GriddedTaskGenerator(storage)
 
+    elif 'geometry' in input_region:  # Larger spatial region
```

Comment:
This is perfectly fine. However, notice that you could merge those
two branches into one:

```python
return GriddedTaskGenerator(storage, geopolygon=geometry, tile_indexes=input_region.get('tiles'))
```

Reply:
fixed.